### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by khalillechelt**
Introduced some config for linting, editorconfig, gitignore etc. 
Also refactored towards es2015 and modularized a bit.

### CODE REVIEW

File: lib/content/gittenify.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import catNames from 'cat-names';
+import mutationSummary from 'mutation-summary';
+import catPhoto from './catPhoto';
+import replaceUsers from './replaceUsers';
+
+export default function gittenify(store) {
+  const userMap = {};
+
+  for (const u of store.users) {
```
Comment (RobinThrift) [inline lib/content/gittenify.js]: `const u` seems strange here. Wouldn't `let` fit better? Semantically.

Comment (khalillechelt) [inline lib/content/gittenify.js]: @RobinThrift `const` should always be used when you know that the variable will not be modified. Actually the eslint config I added errors here when you use `let` because u never changes.

Comment (RobinThrift) [inline lib/content/gittenify.js]: That's what I meant by "Semantically". While the variable inside the block itself doesn't change, the variable "itself" does. If it checks out, it's fine. Just a bit odd ;)

Comment (khalillechelt) [inline lib/content/gittenify.js]: hmm interesting, I guess since you are re-declaring it with a `const` keyword on every turn of that `for`-loop it is like you are just wiping it away and then initializing it fresh every time so that the variable is actually not being changed but erased and re-declared.

Comment (khalillechelt) [inline lib/content/gittenify.js]: or something...

might need some further research.

File: lib/options/ui.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      renderList(users);
+      userInput.value = '';
+    }
+  });
+
+  // handle removals
+  userList.addEventListener('click', (e) => {
+    const t = e.target;
+
+    if (t.className.match(/user-list-remove/)) {
```
Comment (RobinThrift) [inline lib/options/ui.js]: You could pull this RegEx out of the listener to avoid it being created every time. Very nitpicky though ;)

File: lib/content/gittenify.js (hunk 1 above)
Comment (khalillechelt) [inline lib/content/gittenify.js]: Babel doesn't seem to have a problem with it. Good catch though!

Comment (RobinThrift) [inline lib/content/gittenify.js]: Babel transforms this into `for(var u,l=e.users[Symbol.iterator]();!(a=(u=l.next()).done);a=!0)`. I think the variables in the `for`-head only get initialized once, hence why I caught it. I could be wrong though. ES6 sure is interesting ;)

File: lib/options/ui.js (hunk 2 above)
Comment (khalillechelt) [inline lib/options/ui.js]: fixed it :) thx

File: lib/content/replaceUsers.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import lorem from 'lorem-ipsum';
+import { catWords, containerSelector, authorSelector, avatarSelector, commentSelector } from './selectors';
+
+export default function replaceUsers(el, store, userMap) {
+  const containers = el.querySelectorAll(containerSelector);
+
+  [...containers].forEach((container) => {
```
Comment (RobinThrift) [inline lib/content/replaceUsers.js:7]: Great use of the spread operator. I love this ES6 stuff :smile:

File: lib/content/gittenify.js (hunk 1 above)
Comment (khalillechelt) [inline lib/content/gittenify.js]: Interesting. Then the linter is not working correctly in this case I would say.

File: lib/content/replaceUsers.js (hunk 3 above)
Comment (khalillechelt) [inline lib/content/replaceUsers.js:7]: yeah so neat!!!
that's actually es7 as far as I know :))

File: lib/content/gittenify.js (hunk 1 above)
Comment (khalillechelt) [inline lib/content/gittenify.js]: I changed it to let and turned off the `prefer-const` eslint rule :)

File: lib/content/gittenify.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import catNames from 'cat-names';
+import mutationSummary from 'mutation-summary';
+import catPhoto from './catPhoto';
+import replaceUsers from './replaceUsers';
+
+export default function gittenify(store) {
+  const userMap = {};
```
Comment (RobinThrift) [inline lib/content/gittenify.js:7]: We could use a "real" ES6 [`Map`](https://babeljs.io/docs/learn-es2015/#map-set-weak-map-weak-set) here, just for ES6's sake. But I don't know, how that would play with Chrome's storage. Maybe like this: http://www.2ality.com/2015/08/es6-map-json.html

Comment (khalillechelt) [inline lib/content/gittenify.js:7]: good idea, but that is out of scope of this pull request. I just want to clean up and give it some structure. You are very welcome to create a new pull request with that feature added though :smile:

Comment (RobinThrift) [inline lib/content/gittenify.js:7]: Just wanted to mention it for "future reference" ;) Maybe some day I will have time...

**Comment by RobinThrift** [comment]
:+1: Okay, looks good to me. Somebody else might want to check if it works in their browser too. :cat:

File: lib/content/gittenify.js (hunk 4 above)
Comment (khalillechelt) [inline lib/content/gittenify.js:7]: Just open an issue for that and label it with `improvement` or something like that :)

**Comment by khalillechelt** [comment]
@RobinThrift thank you!

**Comment by khalillechelt** [comment]
That's good enough for me right now. I am merging this in :)

**Comment by RobinThrift** [comment]
Okay, great! Cool stuffs!
